src/main/java/novamachina/exnihilosequentia/common/datagen/ExNihiloRecipeGenerator.java

```diff
@@ -356,6 +359,13 @@ private void registerHeatRecipes(Consumer<IFinishedRecipe> consumer) {
         createHeatRecipes(consumer, Blocks.GLOWSTONE, 2, "glowstone");
         createHeatRecipes(consumer, Blocks.SHROOMLIGHT, 2, "shroomlight");
         createHeatRecipes(consumer, Blocks.SOUL_FIRE, 4, "soul_fire");
+
+        // Lightable blocks
+        StatePropertiesPredicate lightable = StatePropertiesPredicate.Builder.properties().hasProperty(BlockStateProperties.LIT, true).build();
```

Comment:
maybe move this in exnihilosequentia/api/datagen/AbstractRecipeGenerator and make a new function for lit recipe (something like "createLitHeatRecipes" just to make it usable in addons)